src/Type/IntrinsicType.ts

```diff
@@ -0,0 +1,23 @@
+import type { BaseType } from "./BaseType.js";
+import { PrimitiveType } from "./PrimitiveType.js";
+
+export class IntrinsicType extends PrimitiveType {
```

Comment:
The term "intrinsic" is not formally defined in the TypeScript documentation, would be good if you write a jsdoc saying that is for generic built in types like `Uppercase` and so on...